### PR TITLE
Include a fresh build in an npm bundle

### LINKED ISSUE #244: Update dist file for every release
Hi, I noticed that the file in /dist is from version 1.3.0 (created form issue #184).
Would it be possible to update it to the latest version and keep it updated with every release? That would be of great help for anyone using virtual-dom and AMD.

### PR DESCRIPTION
**Pull request by tomekwi**
`npm` allows us to include a fresh build in every uploaded version of the package. Without checking in the built files.

We all know that built files in the repo are a nuisance. Whether the builds are manual or automated with some sort of hooks, you still have a lot of crap in the diffs.

Fixes #244

### CODE REVIEW

**Comment by tomekwi** [comment]
:checkered_flag: 

Run `npm pack` to test this out by simulating an `npm publish`.

**Comment by tomekwi** [comment]
I didn’t realize I should have updated your `npm run dist` as well – so that it works in a freshly checked-out repo. @Matt-Esch @Raynos should improve anything else yet?

File: package.json [hunk 1]
```diff
@@ -59,12 +59,13 @@
     "view-cover": "istanbul report html && opn ./coverage/index.html",
     "browser": "run-browser test/index.js",
     "phantom": "run-browser test/index.js -b | tap-spec",
-    "dist": "browserify --standalone virtual-dom index.js > dist/virtual-dom.js",
+    "dist": "mkdir -p dist; browserify --standalone virtual-dom index.js > dist/virtual-dom.js",
```
Comment (neonstalwart) [inline package.json]: shouldn't we be using a package rather than relying on the OS - e.g. windows doesn't have `mkdir`?

Comment (tomekwi) [inline package.json]: Does windows understand bash operators like `|`, `>` and `&&`?

Comment (neonstalwart) [inline package.json]: http://blog.keithcirkel.co.uk/how-to-use-npm-as-a-build-tool/#thewindowsproblem

Comment (tomekwi) [inline package.json]: Thanks for the link! I once looked for binary node packages to replace Unix programs like *cp*, *mv* or *mkdir* – but gave up seeing that no-one really bothers to make them.

I [can’t find](http://npmsearch.com/?q=mkdir) a binary alternative to *mkdir* – so I’ll push a `node --eval` solution.

**Comment by tomekwi** [comment]
:checkered_flag:

File: package.json [hunk 2]
```diff
@@ -59,12 +60,13 @@
     "view-cover": "istanbul report html && opn ./coverage/index.html",
     "browser": "run-browser test/index.js",
     "phantom": "run-browser test/index.js -b | tap-spec",
-    "dist": "browserify --standalone virtual-dom index.js > dist/virtual-dom.js",
+    "dist": "node --eval \"require('mkdirp')('dist')\"; browserify --standalone virtual-dom index.js > dist/virtual-dom.js",
```
Comment (neonstalwart) [inline package.json]:
```json
"dist": "mkdirp dist && browserify ...",
```

**Comment by Zolmeister** [comment]
For reference, I don't think it's worth supporting windows as a build platform

**Comment by tomekwi** [comment]
@Zolmeister now windows support should be there anyway :) At least in `npm run dist` and therefore `npm publish`. I haven’t tested this on an actual box, but nothing is platform-specific there.

**Comment by tomekwi** [comment]
@Matt-Esch @neonstalwart I’ve rebased it into 3 commits, to get rid of the mess. Is this good to go?

**Comment by tomek-he-him** [comment]
Closing to remove clutter 🙂

Feel free to open if you think this is mergeable.

**Pull request closed without merging**